test/new-e2e/pkg/utils/e2e/client/agent_client.go

```diff
@@ -62,3 +70,88 @@ func NewDockerAgentClient(t *testing.T, docker *Docker, agentContainerName strin
 
 	return commandRunner, nil
 }
+
+// waitForAgentsReady waits for the given non-core agents to be ready.
+// The given options configure which Agents to wait for, and how long to wait.
+//
+// Under the hood, this function checks the readiness of the agents by querying their status endpoints.
+// The function will wait until all agents are ready, or until the timeout is reached.
+// If the timeout is reached, an error is returned.
+//
+// As of now this is only implemented for Linux.
+func waitForAgentsReady(tt *testing.T, host *components.RemoteHost, params *agentclientparams.Params) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should this be a method of HostAgentClient instead, and have wait commands initialized at `NewHostAgentClient`?

Reply:
This is not running agent commands so we can't really work from `HostAgentClient` (it only runs agent subcommands) 😕